cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/inference/Inference.kt

```diff
@@ -528,6 +539,52 @@ class Inference internal constructor(val start: Node, override val ctx: Translat
         this.scopeManager = ctx.scopeManager
         this.typeManager = ctx.typeManager
     }
+
+    fun inferReturnType(call: CallExpression?): Type {
+        if (call == null) {
+            return unknownType()
+        }
+
+        // Try to find out, if the supplied hint is part of an assignment.  If yes, we can use their
+        // type as the return type of the function
+        var targetType =
+            ctx.currentComponent.assignments.firstOrNull { it.value == call }?.target?.type
```

Comment:
`firstOrNull` is not really what I'd do here. Since most languages cannot differentiate between the methods to invoke based on the return type, I'd collect everything and use the most specific common supertype

Reply:
The `hint` can only be part of one assignment (if it is part of any) in the AST. Therefore `firstOrNull` is the correct thing to do here. To make it a bit more clearer, I will change it to `singleOrNull`.